community/cypher/cypher-compiler-2.3/src/main/java/org/neo4j/cypher/internal/compiler/v2_3/codegen/CompiledMathHelper.java

```diff
@@ -144,6 +147,90 @@ public static Object subtract( Object lhs, Object rhs )
                                        " and " + rhs.getClass().getSimpleName(), null );
     }
 
+    public static Object multiply( Object lhs, Object rhs )
+    {
+        if ( lhs == null || rhs == null )
+        {
+            return null;
+        }
+
+        if ( lhs instanceof Number && rhs instanceof Number )
+        {
+            if ( lhs instanceof Double || rhs instanceof Double ||
+                 lhs instanceof Float || rhs instanceof Float )
+            {
+                return ((Number) lhs).doubleValue() * ((Number) rhs).doubleValue();
+            }
+            if ( lhs instanceof Long || rhs instanceof Long ||
```

Comment:
Do we support multiplication of `int`, `short`, and `byte`? Is it correct to have it always result in a `long`? In java short * short = int, right?